### DISCUSSION TITLE
Is there any way to convert PDF into picture using ImageSharp?

### DISCUSSION THREAD

**Discussion by GirlWithAnEpee** [SixLabors/ImageSharp#2836] [Q&A] [answered]
So, everything is in the title :) I need to convert a pdf-document into picture (every page should be a picture)? .NET v.4.0, ara there any ways to do it using ImageSharp?

**Comment by antonfirsov** [comment, marked as answer]
PDF rendering is a much wider concern than just imaging, and currently there are no SixLabors implementing it. You would need to look for other libs, eg. https://github.com/QuestPDF/QuestPDF (note that lib has a license similar to ours, if yor comany earns more than $1M, you need a paid license).

**Reply by GirlWithAnEpee** [reply]
Have you ever found any examples how to convert pdf to jpg using it? All I can find is how to create a PDF-document.

**Reply by tiesont** [reply]
Can you clarify which "it" you are referring to? For QuestPDF, that would be https://www.questpdf.com/concepts/generating-output.html#generating-images

**Reply by GirlWithAnEpee** [reply]
I'll try to explain again, sorry, maybe my english is not good enough. As I can see, QuestPDF provides the opportunity to generate a PDF-file and then save it as PDF, image, etc. But  what if the file is not generated with QuestPDF but uploaded for example by a user, how to "ask" QuestPDF to convert it?

**Reply by tiesont** [reply]
Ah. Unfortunately, that's a question to ask on that repo, not here.

**Reply by GirlWithAnEpee** [reply]
Anyway, thanks!

**Comment by tiesont** [comment]
Docnet (https://github.com/GowenGit/docnet) has been the simplest project I've found for exporting PDF pages as images. ImageSharp has had no issues loading and manipulating the bitmaps which Docnet creates.

**Reply by GirlWithAnEpee** [reply]
Unfortunately, it doesn't work with .NET v4.0, which I have to use. :(

**Comment by manfromarce** [comment]
You can try [this project based on PdfPig](https://github.com/BobLd/PdfPig.Rendering.Skia) which is fully managed (no Pdfium). However, it uses SkiaSharp, not ImageSharp.
